torch/csrc/distributed/c10d/NCCLUtils.cpp

```diff
@@ -39,8 +39,21 @@ NCCLComm::NCCLComm(NCCLComm&& other) {
   std::swap(deviceIndex_, other.deviceIndex_);
 }
 
-ncclUniqueId NCCLComm::getNcclId() {
-  return ncclId_;
+void NCCLComm::setUniqueHash(ncclUniqueId ncclId) {
+  const uint8_t* bytes = reinterpret_cast<const uint8_t*>(&ncclId);
+  std::ostringstream oss;
+  for (const auto i : c10::irange(NCCL_UNIQUE_ID_BYTES)) {
+    oss << std::hex << static_cast<int>(bytes[i]);
+  }
+  this->uniqueHash_ = oss.str();
+}
```

Comment:
```suggestion
void NCCLComm::setUniqueHash(ncclUniqueId ncclId) {
  const uint8_t* bytes = reinterpret_cast<const uint8_t*>(&ncclId);
  this->uniqueHash_ = fmt::format("{:02x}", fmt::join(
    bytes, bytes + NCCL_UNIQUE_ID_BYTES, ""
  ));
}
Seems like it might be a bit more efficient (it's pass the hex values so it's unispace though.